engine/lib/notification.php

```diff
@@ -619,34 +618,58 @@ function elgg_send_email($from, $to, $subject, $body, array $params = null) {
 	}
 
 	// strip name from to and from
-
 	$to_address = Address::fromString($result['to']);
 	$from_address = Address::fromString($result['from']);
 
-
+	// make subject
 	$subject = elgg_strip_tags($result['subject']);
 	$subject = html_entity_decode($subject, ENT_QUOTES, 'UTF-8');
 	// Sanitise subject by stripping line endings
 	$subject = preg_replace("/(\r\n|\r|\n)/", " ", $subject);
 	$subject = trim($subject);
-
-	$body = elgg_strip_tags($result['body']);
-	$body = html_entity_decode($body, ENT_QUOTES, 'UTF-8');
-	$body = wordwrap($body);
-
-	$message = new Message();
+	
+	// build email message
+	$message = new EmailMessage();
 	$message->setEncoding('UTF-8');
 	$message->addFrom($from_address);
 	$message->addTo($to_address);
 	$message->setSubject($subject);
-	$message->setBody($body);
+	
+	// make the email body
+	$mime_body = new MimeMessage();
+	
+	// make plaintext message part
+	$plaintext = elgg_view('core/mail/plaintext', ['body' => $result['body']]);
+	if (!empty($plaintext)) {
+		$plain_part = new MimePart($plaintext);
+		$plain_part->setType(Mime::TYPE_TEXT);
+		$plain_part->setCharset('UTF-8');
+		
+		$mime_body->addPart($plain_part);
+	}
+	
+	// make html message part
+	$html = elgg_view('core/mail/html', ['body' => $result['body']]);
```

Comment:
This should probably be a switch somewhere instead. Perhaps a function that takes in a body and returns a MimePart and filters it before with a hook? That way we default to plaintext, and the plugin can create an HTML mime if it wants to.